### PR TITLE
Feature/open parsable files

### LINKED ISSUE #141: See all files of current model
## Description

**As a** user, on the modelization view, when I switch to text view,

**I want** to see all files related to the selected model opened,

**so that** I can quickly edit the files of model.

### LINKED ISSUE #202: Expand folder(s) containing parsed file(s) when switching to Text view
**Describe the bug**

When creating a project or making a change in the Model view of a project, than switching to the Text view, all folders containing the parsed files should be expanded so we can easily find the created/modified files in the file explorer tree.

**To Reproduce**

1. Create a project and a model.
2. Click on `aws ami` component from `terrator-plugin` in the Model view
3. Switch to the Text view
4. Notice that two tabs are open, but the file explorer is not expanded

**Screenshots**

See screenshot to illustrate the problem.
not_wanted

**Expected behavior**

In this case, we want to see the corresponding folders expanded.
wanted

**Application (please complete the following information):**

- leto version [1.0.0]

### PR DESCRIPTION
**Pull request by k-y0u**
Expand model folders and open parsable files.

close #141, fix #202

### CODE REVIEW

File: src/components/FileExplorer.vue [hunk 1]
```diff
@@ -152,11 +158,55 @@ function onNodeDoubleClicked(node) {
...
+}
+
 /**
  * Update file explorer tree nodes.
+ * Also, expand model folder and open parsable files.
  */
 function updateFileExplorer() {
   nodes.value = getTree(props.projectName, props.fileInformations);
+
+  // Didn't a proper way to stub it on shallowMount.
```
Comment (justinehell) [inline src/components/FileExplorer.vue:200]: `Didn't find a proper way...`
Also why specify this here ?

File: tests/unit/components/FileExplorer.spec.js [hunk 2]
```diff
@@ -270,6 +281,41 @@ describe('Test component: FileExplorer', () => {
...
+      wrapper.vm.openParsableFiles({
+        children: [{ id: 'filename.ext' }],
+      });
+
+      expect(wrapper.vm.activeFileId).toEqual('filename.ext');
+    });
+  });
+
+  describe('Test function: expandFolders', () => {
+    it('should call getPluginByName when last of folder have been expanded', () => {
```
Comment (justinehell) [inline tests/unit/components/FileExplorer.spec.js:300]: `has been`

File: src/components/FileExplorer.vue [hunk 3]
```diff
@@ -152,11 +158,55 @@ function onNodeDoubleClicked(node) {
...
+      expandFolders(folders.slice(1));
+    } else {
+      openParsableFiles(node);
+    }
+  }
+}
+
 /**
  * Update file explorer tree nodes.
+ * Also, expand model folder and open parsable files.
```
Comment (justinehell) [inline src/components/FileExplorer.vue:195]: `model folders`

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/components/FileExplorer.vue (hunk 1 above)
Comment (k-y0u) [reply src/components/FileExplorer.vue:200]: because the if under is only to pass the test. To Delete if another solution is found.

Comment (k-y0u) [reply src/components/FileExplorer.vue]: done

File: tests/unit/components/FileExplorer.spec.js (hunk 2 above)
Comment (k-y0u) [reply tests/unit/components/FileExplorer.spec.js]: done

File: src/components/FileExplorer.vue (hunk 3 above)
Comment (k-y0u) [reply src/components/FileExplorer.vue]: done

**Review by Zorin95670 — DISMISSED** [review]

File: src/components/FileExplorer.vue [hunk 4]
```diff
@@ -247,8 +253,73 @@ async function updateAllFilesStatus() {
...
+ * Expand folder tree nodes corresponding to the selected model.
+ * Only expand the ones containing parsable file tree nodes.
+ * Then, open those file tree nodes.
+ */
+function openModelFiles() {
+  const [pluginName, modelName] = query.value.path.split('/');
+
+  expandFolder(props.projectName);
+
+  if (pluginName && pluginName.length > 0) {
```
Comment (Zorin95670) [inline src/components/FileExplorer.vue:291]: `pluginName?.length > 0`

File: src/components/FileExplorer.vue [hunk 5]
```diff
@@ -247,8 +253,73 @@ async function updateAllFilesStatus() {
...
+    expandFolder(pluginName);
+
+    if (modelName && modelName.length > 0) {
```
Comment (Zorin95670) [inline src/components/FileExplorer.vue:294]: `modelName?.length > 0`

File: src/components/FileExplorer.vue [hunk 6]
```diff
@@ -257,6 +328,11 @@ async function initTreeNodes() {
   nodes.value = getTree(props.projectName, localFileInformations.value);
 
   await updateAllFilesStatus();
+
+  // Didn't find a proper way to stub it on shallowMount.
```
Comment (Zorin95670) [inline src/components/FileExplorer.vue:332]: `// TODO: find a better way to stub it on shallowMount`

File: changelog.md [hunk 7]
```diff
@@ -22,6 +22,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * Modelization feature:
   * Allow to use the mouse coordinates when placing new component on drag and drop feature.
 * Add new environment variable `KEEP_CYPRESS_ATTRIBUTE` to keep all cypress attribute in html.
+* Expand model folders and open parsable files.
```
Comment (Zorin95670) [inline changelog.md:25]: - Move this in `Text editor feature`
- change it by something like this:
`On open text editor open all related files of the current model`
- Add in `### Fixed` section the bug fix you do

**Review by justinehell — CHANGES_REQUESTED** [review]

File: src/components/FileExplorer.vue (hunk 4 above)
Comment (k-y0u) [reply src/components/FileExplorer.vue]: done

File: src/components/FileExplorer.vue (hunk 5 above)
Comment (k-y0u) [reply src/components/FileExplorer.vue]: done

File: src/components/FileExplorer.vue (hunk 6 above)
Comment (k-y0u) [reply src/components/FileExplorer.vue]: done

File: changelog.md (hunk 7 above)
Comment (k-y0u) [reply changelog.md]: done

File: tests/unit/components/FileExplorer.spec.js [hunk 8]
```diff
@@ -453,6 +464,79 @@ describe('Test component: FileExplorer', () => {
...
+
+      wrapper.vm.openFile('file');
+
+      expect(wrapper.vm.activeFileId).toEqual('fileId');
+      expect(selectFileNodeEventNext).toHaveBeenCalledTimes(1);
+    });
+  });
+
+  describe('Test function: openModelFiles', () => {
+    it('should call getPluginByName() otherwise', () => {
```
[the remark this answers is not in the document]
Comment (k-y0u) [reply tests/unit/components/FileExplorer.spec.js]: Done

**Review by justinehell — APPROVED** [review]

**Pull request merged**